gpii/node_modules/flowManager/src/FlowManager.js

```diff
@@ -358,6 +381,56 @@ gpii.flowManager.local.savePreferences = function (that, gpiiKey, preferences) {
     }
 };
 
+/**
+ * Retrieve the revision of the repository from the CBFM and construct the full
+ * URL of the associated solutions registry document
+ * @param {Component} that - An instance of gpii.flowManager.local.
+ * @param {Component} that.gpiiRevisionRequester - An instance of
+ *      gpii.flowManager.revisionRequester used to access the Cloudbased
+ *      FlowManager's "/revision" endpoint.
+ * @param {Component} that.deviceReporter - An instance of
+ *      gpii.deviceReporter used to for the platform information.
+ * @param {Component} that.repositorySolutionsLoader - An instance of
+ *      gpii.flowManager.repositorySolutionsLoader that does the actual fetch.
+ * @param {Component} that.solutionsRegistryDataSource - An instance of
+ *      kettle.dataSource whose fullSolutionsRegistry member will be set to the
+ *      solutions registry fetched by the repositorySolutionsLoader, on success.
+ *      If repositorySolutionsLoader is unsuccessful, the
+ *      solutionsRegistryDataSource is left untouched.
+ * @param {Event} loadedFromRepositoryEvent - Event to fire when done loading,
+ *      even on failure.
+ * @return {Promise} - A promise: when resolved with no error, loads the relevant
+ *      solutions registry file from the source code repository.  It is stored
+ *      in the respositorySolutionsLoader's solutionsRegistry member.
+ */
+gpii.flowManager.local.loadSolutionsFromRepository = function (that, loadedFromRepositoryEvent) {
```

Comment:
As said earlier, this logic probably makes more sense to be inside `solutionRegistryDataSource` component.

Also, by looking at nested promises in here, please consider to use [fluid.promise.fireTransformEvent()](https://docs.fluidproject.org/infusion/development/PromisesAPI.html#fluidpromisefiretransformeventevent-payload-options) that can chain up asynchronous promises and pass the output of the previous promise to the next one on the chain. 

In the `solutionRegistryDataSource` component, the sequence of actions performed by a chain could be:
1. fetch the revision from remote CBFM;
2. Use the revision to fetch SR from Github;
3. If SR from 2 is not null, pass the SR to 4; if it is null, read SR from the local disk and pass to 4;
4. If SR is not null, fire the success event. Otherwise, call `fluid.fail()`.